src/codewhisperer/activation.ts

```diff
@@ -295,8 +300,14 @@ export async function activate(context: ExtContext): Promise<void> {
                                 CodeWhispererConstants.accessTokenMigrationDoNotShowAgainKey,
                                 true
                             )
+                        } else if (resp === CodeWhispererConstants.accessTokenMigrationLearnMore) {
+                            await vscode.commands.executeCommand('aws.codeWhisperer.accessTokenErrorLearnMore')
                         }
                     })
+                context.extensionContext.globalState.update(
+                    CodeWhispererConstants.accessTokenMigrationDoNotShowLastShown,
+                    Date.now()
+                )
```

Comment:
Here we are reusing these `accessTokenMigrationDoNotShowLastShown` and `accessTokenMigrationDoNotShowAgainKey` which are currently used for the warningMessage (before 1/31). So one issue it can happen is that the user closes the message on 1/30 and the next error message cannot show until 2/6(7 days after), which is not ideal(and similar for the `do not show again" again) Ideally they should have different "do not show again" status and different 7 days trackers.

Reply:
Another thing is all of these logic are currently under `if (AuthUtil.instance.hasAccessToken()) {` on line 226. But this boolean is no longer true after the first time it upgrades the connection. So next time(7 days after) it will not enter this if block. Try to come up with a condition that you know this user previously has accessToken and didn't upgrade the connection and simply closed the error notification(without checking "do not show again").

Reply:
makes sense, will update

Reply:
@andrewyuq updated to use different states. For the second issue you mentioned. `AuthUtil.instance.hasAccessToken()` checks if there is a token string stored in the state, this state only gets cleared when user successfully connect to Sono/SSO. So if user had access token and closes the error popup without clicking "Don't show again", they will see the popup after 7 days. If user had access token but choose to connect with Sono/SSO, they won't see this popup ever again. Let me know if this is the expected behavior. Thanks!

Reply:
So we are saying, the accessToken is still there on-disk for the user, but we clear the accessToken connection after 1/31 so they will remain logged out but the accessToken will still be there until user connects to AWS Builder ID?

Reply:
actually the accessToken will also get cleared if the user try to invoke the service using the token and got back "AccessDenied" error, and then they will not see the popup and maybe that's not ideal. Will sync with you offline.

Reply:
discussed offline, we will first nullify accessToken on 1/31 and set another flag to track if previously had an accessToken to cover all cases